### PR TITLE
Add TextInput Filled and Outlined components

### LINKED ISSUE #2: Create TextInput filled

### PR DESCRIPTION
**Pull request by GerardPaligot**
## Changes description 🧑‍💻

* Create `text-inputs` module for TextInput Filled and Outlined components
* Prepare all modules to be published on Maven Central.

## Context 🤔

Closes #2 #3 #15

## Checklist ✅

 - [x] I have reviewed the submitted code
 - [x] I have tested on a phone device/emulator
 - [x] I have tested on a tablet device/emulator
 - [x] I have tested on a kiosk device/emulator

#### Phone
<details>

</details>

#### Tablet
<details>

</details>

#### Kiosk
<details>

</details>

### CODE REVIEW

File: text-inputs/src/main/res/values/styles-text-inputs.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="Widget.Vitamin.TextInputLayout" parent="Widget.Vitamin.TextInputLayout.FilledBox" />
```
Comment (MaximeJineMaraval) [inline text-inputs/src/main/res/values/styles-text-inputs.xml:4]: Maybe add regions for Filled and Outline to improve file readability?

**Review by MaximeJineMaraval — CHANGES_REQUESTED** [review]
I had two comments in the code. I have also three more comments here : 

1. I don't see documentation for the text-inputs. Prefer add it directly with the component's PR to be ready to use in production 😉 
2. What do you think of fixing [this issue](https://github.com/Decathlon/vitamin-android/issues/15) directly in this PR ? It will fixe the full white snackbar when you click on the calendar icon in the sample screen.
3. Do we need a validation from the design team before merge ?

File: sample/build.gradle [hunk 2]
```diff
@@ -26,7 +30,7 @@ android {
 }
 
 dependencies {
-    implementation project(":vitamin")
+    implementation "com.decathlon.vitamin:vitamin:1.0.0-SNAPSHOT"
```
[the remark this answers is not in the document]
Comment (GerardPaligot) [reply sample/build.gradle:33]: Mistake from my side during my tests to fix multidex bug.

File: text-inputs/src/main/res/values/styles-text-inputs.xml (hunk 1 above)
Comment (GerardPaligot) [reply text-inputs/src/main/res/values/styles-text-inputs.xml:4]: We defined `Widget.Vitamin.TextInputLayout.OutlinedBox` and `Widget.Vitamin.TextInputLayout.FilledBox` but this one is the default style applied for style attribute `textInputStyle`.

File: text-inputs/README.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+implementation("com.decathlon.vitamin:vitamin:<version>")
+```
+
+```xml
+<androidx.constraintlayout.widget.ConstraintLayout
+         android:layout_width="match_parent"
+         android:layout_height="wrap_content"
+         android:theme="@style/Theme.Vitamin">
+
+    <com.google.android.material.textfield.TextInputLayout
```
Comment (MaximeJineMaraval) [inline text-inputs/README.md:173]: Prefer use `com.decathlon.android.textinput.VitaminTextInputLayoutFilled`, nope ?

**Review by MaximeJineMaraval — CHANGES_REQUESTED** [review]

**Comment by GerardPaligot** [comment]
@Sabrinavigil That could be great to have a design review here!

**Review by MaximeJineMaraval — APPROVED** [review]

**Pull request merged**